src/storage/ipfsStore.ts

```diff
@@ -18,10 +21,28 @@ export class IPFSSTORE<CID = any, IPFSCLIENT = any>
     public cid: CID;
     public directory: any;
     public ipfsClient: any;
+    public loader: any;
+    public hamt: boolean;
+    public key: string;
 
-    constructor(cid: CID, ipfsClient: IPFSCLIENT) {
+    constructor(cid: CID, ipfsClient: any) {
         this.cid = cid;
+        this.hamt = false;
         this.ipfsClient = ipfsClient;
+        this.key="";
+        this.loader = {
+            async get(cid: CID) {
+                const bytes = await ipfsClient.block.get(cid, {
+                    codec: "dag-cbor",
+                });
+                return bytes;
+            },
+            // For our purposes of reading the HashMap we don't need to implement put
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            async put(cid: CID, bytes: ArrayBuffer) {
```

Comment:
hmm maybe we should support the put anyway for completion of the PR (esp when we plan to push changes upstream) and there is support for it  which does not seem the most complex https://github.com/rvagg/js-ipld-hashmap. 

Worst case maybe not for this PR but maybe definitely think we should support it.

Reply:
put isn't all clear to me as storing bytes on ipfs will generate a new CID so i am not sure why we would need to pass in a CID with the bytes

Reply:
ah so he has it as a CID but in reality it's the "key" based on the docs here https://github.com/rvagg/js-ipld-hashmap/blob/48dbab784a2b845d12aedd4b0bb01ba922a2bd8e/README.md#async-hashmapimplcreateloader-options . I'm guessing we can just support the same implementation though we don't use it for future proofing?